### PR TITLE
[Backport 3.6] PSA: use static key slots to store keys

### PR DESCRIPTION
**Pull request by valeriosetti**
## Description

This is the backport of #9302 to the 3.6 LTS branch. 

## PR checklist

- [x] **changelog** provided
- [x] **development PR** provided #9302 
- [x] **framework PR** not required
- [x] **3.6 PR** here
- [x] **2.28 PR** not required (new feature, no bug fix)
- **tests**  provided

### CODE REVIEW

**Comment by valeriosetti** [comment]
I removed the `need-ci` label because the OpenCI is fine. Failures were only found in the InternalCI (which I cannot check), but since they are not confirmed by the OpenCI, I think they might be spurious failures.

**Comment by tom-cosgrove-arm** [comment]
Note: conflicts

**Comment by gilles-peskine-arm** [comment]
Now that the main PR has been approved, please complete and rebase this backport.

**Comment by valeriosetti** [comment]
I've just backported all the commits from #9302 to `mbedtls-3.6`. Only some of them required a manual fix, in particular a couple of `*.data` files (pkwrite and pkparse) that were moved in `tf-psa-crypto` in the `development` branch.

File: tests/suites/test_suite_pkwrite.data
Comment (tom-cosgrove-arm) [inline tests/suites/test_suite_pkwrite.data:11]: How come this line needs `../../framework` but line 7 only needs `../framework`?

Comment (valeriosetti) [reply tests/suites/test_suite_pkwrite.data:11]: Ops, sorry, that's one of the 2 files who had conflicts during the rebase and apprently I missed a change here :/

**Review by tom-cosgrove-arm — APPROVED** [review]
I've compared the diffs for each of the files in the two PRs - test_suite_pkparse.data had  by far the most changes - and I am happy that this is a proper backport of #9302

**Comment by valeriosetti** [comment]
CI reported a lot of failures and I found that this was due to the `framework` not being properly updated. So:
- I rebased https://github.com/Mbed-TLS/mbedtls-framework/pull/48 on top of `framework` `main`
- I added a new (cherry-picked) commit to make use of the new `framework` reference

**Review by tom-cosgrove-arm — DISMISSED** [review]
LGTM (subject to CI, of course)

**Review by gilles-peskine-arm — DISMISSED** [review]
LGTM

This is a faithful backport of https://github.com/Mbed-TLS/mbedtls/pull/9302 and I can't think of anything that would need to be different. In particular, I am satisfied that there is no change in behavior, and no non-trivial change in code size, when the new option `MBEDTLS_PSA_STATIC_KEY_SLOTS` remains disabled.

**Review by gilles-peskine-arm — DISMISSED** [review]
LGTM after rebase from 3558cba6504897e9f7dc4ecd5766c5cc2abff3e9 to update the framework

**Comment by valeriosetti** [comment]
@bensze01 I'm adding you as reviewer for this one since it's the backport of #9302, where you are a reviewer, and it simply cherry-picks commits from there so it should be relatively fast to review.

**Comment by valeriosetti** [comment]
I just backported the same changes done in `development` https://github.com/Mbed-TLS/mbedtls/pull/9302#issuecomment-2429027625

**Review by gilles-peskine-arm — APPROVED** [review]
Looks good to me after rebase from [my previous approval](https://github.com/Mbed-TLS/mbedtls/pull/9448#pullrequestreview-2372234717) at b287a4056d5639bb58743de924452a7d596a9f05 to the current head at 40859ac3b6ad62fccbd5078207480515b654bdbd.

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM

**Pull request merged**
